dashboard/agent.py

```diff
@@ -385,14 +398,20 @@ async def _check_parent():
             # Agent is failed to be started many times.
             # Push an error to all drivers, so that users can know the
             # impact of the issue.
-            redis_client = ray._private.services.create_redis_client(
-                args.redis_address, password=args.redis_password)
+            redis_client = None
             gcs_publisher = None
-            if args.gcs_address:
-                gcs_publisher = GcsPublisher(args.gcs_address)
-            elif gcs_pubsub_enabled():
-                gcs_publisher = GcsPublisher(
-                    address=get_gcs_address_from_redis(redis_client))
+            if gcs_pubsub_enabled():
+                if use_gcs_for_bootstrap():
```

Comment:
aside: two feature flags is a bit confusing, it would have been nicer to have just one golden path. But it's fine for now.